scripts/babel/extension-resolver.js

```diff
@@ -0,0 +1,145 @@
+// original source: https://github.com/shimataro/babel-plugin-module-extension-resolver
+// To create proper ES Modules, the paths imported need to be fully-specified,
+// and can't be resolved like is possible with CommonJS. To have large compatibility
+// without much hassle, we don't use extensions *inside* the source code, but add
+// them with this plugin.
+
+const fs = require('fs');
+const path = require('path');
+
+const PLUGIN_NAME = 'babel-plugin-extension-resolver';
+
+// InstantSearch.js/es is an ES Module, so needs complete paths,
+// React-DOM also fails if the paths are incomplete
+// For verification, see test/module/packages-are-es-modules.mjs
+const importStartsWith = ['instantsearch.js/', 'react-dom/'];
```

Comment:
Actually, can we pass these imports to the `extensionResolver({ imports: ['instantsearch.js', 'react-dom'] })` call in the Babel config? (and add the `/` in this plugin)

It makes more sense to decouple this for the plugin itself to remain generic.

Reply:
yes, could do, although the slash is specifically because we only want non-global imports to have extension added. I guess that can be generalised indeed